### PR TITLE
Add unit testing support for cache-creator

### PR DESCRIPTION
**Pull request by kuhar**
Import the [doctest](https://github.com/onqtam/doctest) unit testing
framework and integrate it with the build system.

### CODE REVIEW

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by dstutt** [comment]
This seems fine to me.

File: tools/cache_creator/CMakeLists.txt [hunk 1]
```diff
@@ -64,4 +64,11 @@ target_link_libraries(cache-creator PRIVATE cache_creator_lib)
 # Build cache-creator whenever we build XGL.
 add_dependencies(xgl cache-creator)
 
-# TODO: Add unit tests.
+# This executable cache-creator runs unit tests.
```
Comment (kuhar) [inline tools/cache_creator/CMakeLists.txt:67]: nit: should be `This executable runs cache-creator unit tests.`

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 785f315d44d60e1e3f8edd25344b24e1a65a3200
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev f2c529ec94dd68361184fa712ad2aded5abdd36b</li>
<li>PAL: dev 1e61f9ab6eb2041c73d7e7bd8f2021fa05a4bdd8</li>
<li>SPVGEN: dev 7c1cf34dd7eb4d3a6e6652c26292eef1bfcbcaf0</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1d0178f3d37675e22bf1a16267d1d070fcb8d720</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/224697)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26847/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26847/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25436/56232 (45.2%)</li>
<li>Failed: 0/56232 (0.0%)</li>
<li>Skipped: 30796/56232 (54.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26862/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29293/56155 (52.2%)</li>
</ul></details>


</ul></details>

**Comment by kuhar** [comment]
V1: Fix the comment.

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit ef8a9be77fb2e6a51d495c88c0695203d4cbe1ef
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev f2c529ec94dd68361184fa712ad2aded5abdd36b</li>
<li>PAL: dev 1e61f9ab6eb2041c73d7e7bd8f2021fa05a4bdd8</li>
<li>SPVGEN: dev 7c1cf34dd7eb4d3a6e6652c26292eef1bfcbcaf0</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1d0178f3d37675e22bf1a16267d1d070fcb8d720</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/224697)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26847/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26847/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25436/56232 (45.2%)</li>
<li>Failed: 0/56232 (0.0%)</li>
<li>Skipped: 30796/56232 (54.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26862/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29293/56155 (52.2%)</li>
</ul></details>


</ul></details>

**Review by dstutt — APPROVED** [review]
This LGTM

**Comment by JaxLinAMD** [comment]
@kuhar  please resolve conflicts

**Comment by kuhar** [comment]
> @kuhar please resolve conflicts

Done. Code formatting check fails because of misused relative paths, but we should ignore it for this PR. I'll address the formatting check issues in a separate PR.

**Comment by JaxLinAMD** [comment]
retest this please

**Pull request merged**